### PR TITLE
feat: script receiver

### PR DESCRIPTION
**Pull request by weliasz**

### CODE REVIEW

**Comment by rmfitzpatrick** [comment]
Thanks for the contribution. Can you please break this up into different PRs, the first being the component skeleton w/ the readme? I think it would make sense to have individual PRs for each added script, and I'd think that shellcheck or similar linting tests should be added.

Incorporation into the distribution as a component would be another PR, and any incidental changes to our builds/dockerfiles would be separate.

https://github.com/signalfx/splunk-otel-collector/blob/main/CONTRIBUTING.md#contributing-via-pull-requests

File: internal/signalfx-agent/bundle/Dockerfile [hunk 1]
```diff
@@ -43,7 +43,7 @@ RUN ENCODED_VER=$(echo $JDK_VERSION | sed 's/_/%2B/g') && \
     else \
     OPENJDK_URL="${OPENJDK_BASE_URL}/jdk-${ENCODED_VER}/OpenJDK11U-jdk_${ARCH}_linux_hotspot_${JDK_VERSION}.tar.gz"; \
     fi && \
-    wget -nv -O /tmp/openjdk.tar.gz "$OPENJDK_URL"
+    wget -nv --no-check-certificate -O /tmp/openjdk.tar.gz "$OPENJDK_URL"
```
Comment (atoulme) [inline internal/signalfx-agent/bundle/Dockerfile:46]: Please revert this change and other occurrences of that change, as we want to check certs for our downloads as much as possible.

File: internal/signalfx-agent/bundle/scripts/get-collectd-plugins.py [hunk 2]
```diff
@@ -60,7 +62,7 @@
     requirements_file = os.path.join(plugin_dir, "requirements.txt")
     if os.path.isfile(requirements_file):
         subprocess.check_call(
-            [PYTHON_EXECUTABLE, "-m", "pip", "install", "-qq", "--no-warn-script-location", "-r", requirements_file]
+            [PYTHON_EXECUTABLE, "-m", "pip", "install", "--trusted-host", "github.com", "--trusted-host", "codeload.github.com", "-qq", "--no-warn-script-location", "-r", requirements_file]
```
Comment (atoulme) [inline internal/signalfx-agent/bundle/scripts/get-collectd-plugins.py:65]: What is the impetus for this change?

File: internal/version/version_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+package version
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSettingsWithHelpFlags(t *testing.T) {
+	require.Equal(t, 2, 2)
```
Comment (atoulme) [inline internal/version/version_test.go:25]: Please consider deleting that test file.

File: tmp/go.sum [hunk 4]
```diff
@@ -0,0 +1,480 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (atoulme) [inline tmp/go.sum:1]: This file can be deleted.

File: tmp/go.mod [hunk 5]
```diff
@@ -0,0 +1,61 @@
+module github.com/signalfx/splunk-otel-collector/internal/receiver/scriptreceiver
```
Comment (atoulme) [inline tmp/go.mod:1]: Please delete that file.

File: internal/receiver/scriptreceiver/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
+# Script Receiver
+
+<!-- status autogenerated section -->
+| Status        |                        |
+| ------------- |------------------------|
+| Stability     | [In Development]: logs |
+| Distributions | [contrib]              |
+
+[beta]: https://github.com/open-telemetry/opentelemetry-collector#beta
```
Comment (atoulme) [inline internal/receiver/scriptreceiver/README.md:9]: This should match the stability line.

File: cmd/otelcol/config/collector/full_config_linux.yaml [hunk 7]
```diff
@@ -167,6 +167,201 @@ receivers:
   fluentforward:
     endpoint: 0.0.0.0:8006
 
+  # Enables filelog receiver
```
Comment (atoulme) [inline cmd/otelcol/config/collector/full_config_linux.yaml:170]: this needs to be a separate PR

**Comment by atoulme** [comment]
Can we please have tests run to check the behavior of this receiver to make sure that each script is able to run at least on linux amd64?

**Comment by weliasz** [comment]
Thanks for quick comments. I'd definitely remove certification verification bypassing logic as it failed my local build an by mistake ended up on the PR. This not final version so some garbage remained which I'll clean immediately. I wanted to get some feedback quickly to know if any additional changes are required. I'm also working on adding unit and integration tests.

**Comment by atoulme** [comment]
No problem. Appreciate the great start, keep pushing :)

**Comment by weliasz** [comment]
@atoulme - could you point me to some existing testing solutions in splunk OTEL distribution that would allow to run integrations tests on linux amd64 specificlly.

**Comment by rmfitzpatrick** [comment]
> @atoulme - could you point me to some existing testing solutions in splunk OTEL distribution that would allow to run integrations tests on linux amd64 specificlly.

The integration tests are here and there is some relatively thorough documentation: https://github.com/signalfx/splunk-otel-collector/tree/main/tests. An example logging test: https://github.com/signalfx/splunk-otel-collector/blob/main/tests/receivers/filelog/filelog_test.go
